### PR TITLE
fix(wrangler): Rename more `legacy-assets` things

### PR DESCRIPTION
**Pull request by CarmenPopoviciu**
## What this PR solves / how to test

[Workers + Assets]

This PR renames a whole bunch of `assets` things to `legacy-assets`

🚨 Pls review carefully as this touches many parts of the codebase

## Author has addressed the following

- Tests
  - [ ] TODO (before merge)
  - [ ] Included
  - [x] Not necessary because: our current tests should cover us
- E2E Tests CI Job required? (Use "e2e" label or ask maintainer to run separately)
  - [ ] I don't know
  - [x] Required / Maybe required
  - [ ] Not required because:
- Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))
  - [ ] TODO (before merge)
  - [ ] Included
  - [x] Not necessary because: just naming changes
- Public documentation
  - [ ] TODO (before merge)
  - [ ] Cloudflare docs PR(s): 
  - [x] Not necessary because: just naming changes

### CODE REVIEW

**Review by RamIdeas — APPROVED** [review]
I don't see any logic changes in this PR so I can confidently say if typescript checks pass then this LGTM!

File: packages/wrangler/src/dev/local.tsx [hunk 1]
```diff
@@ -89,7 +89,8 @@ export async function localPropsToConfigBundle(
 		inspectorPort: props.runtimeInspectorPort,
 		bindings: props.bindings,
 		workerDefinitions: props.workerDefinitions,
-		assetPaths: props.assetPaths,
+		// TODO @Carmen can we rename here?
```
Comment (petebacondarwin) [inline packages/wrangler/src/dev/local.tsx:92]: What is the answer here?

**Review by petebacondarwin — COMMENTED** [review]
Should this have a changeset? I get that it is an internal refactor, so perhaps not?

File: packages/wrangler/src/dev/local.tsx (hunk 1 above)
Comment (CarmenPopoviciu) [reply packages/wrangler/src/dev/local.tsx:92]: the answer was yes, and `assetsPath` was renamed to `legacyAssetsPath`, but thank you for spotting the todo comment. I cleaned that up

**Comment by CarmenPopoviciu** [comment]
> Should this have a changeset? I get that it is an internal refactor, so perhaps not?

This was my reasoning as well. These are not user facing changes, since all we're doing is renaming some internal things. The user-facing nomenclature already happened in a separate PR

**Pull request merged**